Cabal/Distribution/Simple/GHC.hs

```diff
@@ -772,6 +768,9 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
   -- TODO: problem here is we need the .c files built first, so we can load them
```

Comment:
what about this todo?

Sadly it looks like that we don't have test for that. 

But this TODO makes me worry what we might break.

Reply:
This comment was here before a6e427a, so presumably it wasn't fixed given that c files that didn't depend on the ffi generated .h files didn't even load. As far as I can tell, nothing that depended on a c source would have worked, so there's nothing really to break

Reply:
It seems the comment is talking about this "feature": https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/ffi.html#using-foreign-export-and-foreign-import-ccall-wrapper-with-ghc

The C sources could be depending on these `_stub.h` headers GHC generates for `foregin export` decls.